src/java/org/apache/cassandra/tools/nodetool/ConsensusMigrationAdmin.java

```diff
@@ -0,0 +1,111 @@
+{
+    @Command(name = "list", description = "List migrating tables and ranges")
+    public static class ListCmd extends ConsensusMigrationAdmin
+    {
+        @Arguments(usage = "[<keyspace> <tables>...]", description = "The keyspace followed by one or many tables")
+        private List<String> schemaArgs = new ArrayList<>();
+
+        @Option(title = "format", name = {"-f", "--format"}, description = "Output format, YAML and JSON are the only supported formats, default YAML, prefix with `minified-` to turn off pretty printing")
+        private String format = "yaml";
+
+        protected void execute(NodeProbe probe)
+        {
+            Set<String> keyspaceNames = schemaArgs.size() > 0 ? singleton(schemaArgs.get(0)) : null;
+            Set<String> tableNames = schemaArgs.size() > 1 ? new HashSet<>(schemaArgs.subList(1, schemaArgs.size())) : null;
+            String output = probe.getStorageService().listConsensusMigrations(keyspaceNames, tableNames, format);
+            probe.output().out.println(output);
+        }
+    }
+
+    @Command(name = "begin-migration", description = "Mark the range as migrating for the specified token range and tables")
+    public static class BeginMigration extends ConsensusMigrationAdmin
+    {
+        @Option(title = "start_token", name = {"-st", "--start-token"}, description = "Use -st to specify a token at which the repair range starts")
+        private String startToken = null;
+
+        @Option(title = "end_token", name = {"-et", "--end-token"}, description = "Use -et to specify a token at which repair range ends")
+        private String endToken = null;
+
+        @Option(title = "target_protocol", name = {"-tp", "--target-protocol"}, description = "Use -tp to specify what consensus protocol should be migrated to", required=true)
+        private String targetProtocol = null;
+
+        @Arguments(usage = "[<keyspace> <tables>...]", description = "The keyspace followed by one or many tables")
+        private List<String> schemaArgs = new ArrayList<>();
+
+        protected void execute(NodeProbe probe)
+        {
+            checkNotNull(startToken, "Start token must be specified");
+            checkNotNull(endToken, "End token must be specified");
+            checkArgument(schemaArgs.size() >= 2, "Must specify a keyspace and at least one table");
+            checkArgument((endToken != null && startToken != null) || (endToken == null && startToken == null), "Must specify start and end token together");
+            String maybeRangesStr = startToken != null ? startToken + ":" + endToken : null;
+            List<String> keyspaceNames = schemaArgs.size() > 0 ? singletonList(schemaArgs.get(0)) : emptyList();
+            List<String> maybeTableNames = schemaArgs.size() > 1 ? schemaArgs.subList(1, schemaArgs.size()) : null;
+            probe.getStorageService().migrateConsensusProtocol(targetProtocol, keyspaceNames, maybeTableNames, maybeRangesStr);
```

Comment:
We should output a message indicating that the specified tables/ranges need to be repaired to complete migration

Reply:
Added a message although hopefully we automate so this can be removed.